zson/marshal.go

```diff
@@ -680,6 +680,13 @@ func (u *UnmarshalZNGContext) decodeAny(zv zed.Value, v reflect.Value) error {
 		v.Set(reflect.ValueOf(*zv.Copy()))
 		return nil
 	}
+	if typ := zed.TypeUnder(zv.Type); typ == zed.TypeNull && v.Kind() != reflect.Ptr {
+		// Since zed.TypeNull can be any type, set the interface to zero value.
+		// We need to make sure the reflect value isn't a pointer since pointers
+		// structs on null values should remain a nil pointer.
```

Comment:
I don't follow this. Can you elaborate on "pointers structs on null values should remain a nil pointer"?

Reply:
Yeah I'm going to do another pass of comments on this logic because it is confusing.

Reply:
Updated, I hope this is clearer.